modules/profile/main.tf

```diff
@@ -0,0 +1,54 @@
+data "ibm_scc_control_libraries" "scc_control_libraries" {
+  instance_id = var.instance_id
+}
+
+locals {
+  control_libraries = flatten([for control_library in data.ibm_scc_control_libraries.scc_control_libraries.control_libraries : [
+    for ctrl in var.control_libraries :
+    control_library if ctrl.control_library_name == control_library.control_library_name && ctrl.control_library_version == control_library.control_library_version
+    ]
+  ])
+}
+
+data "ibm_scc_control_library" "scc_control_library" {
+  count              = length(var.control_libraries)
+  instance_id        = var.instance_id
+  control_library_id = local.control_libraries[count.index].id
+}
+
+locals {
+  controls_map = flatten([
+    for index, control_library in local.control_libraries : [
+      for control in data.ibm_scc_control_library.scc_control_library[index].controls : {
+        control_library_id = control_library.id
+        control_id         = control.control_id
+      }
+    ]
+  ])
+}
+
+resource "ibm_scc_profile" "scc_profile_instance" {
+  instance_id         = var.instance_id
+  profile_description = var.profile_description
+  profile_name        = var.profile_name
+  profile_type        = "custom"
+
+  dynamic "controls" {
+    for_each = local.controls_map
+    content {
+      control_library_id = controls.value.control_library_id
+      control_id         = controls.value.control_id
```

Comment:
When creating a custom profile, 2 different types of controls can be added:
- A predefined control from a specific control library
- A custom control that you may of created yourself

Seems this code is only supporting the predefined approach?

Reply:
I think the module needs to be able to do what a user can do with in the UI. A control library is not required to create a custom profile. All that is required is controls

Reply:
I've updated it to add individual rules, let me know if this is more what it should look like. I can add a flag such as `add_all_profile_controls` or something along those lines if the user wants everything.

Reply:
It still asks for a control library because you need to supply a control library that the control exists in.

Reply:
ok yes you are right - and this is the same in the UI - you can only add a control from a control library